tests/coverage/coverage.cpp

```diff
@@ -756,6 +756,88 @@ TEST(Importer, notMathMLMathNodesInComponentMath)
     EXPECT_EQ("Units reference 'daves' in units 'bobs' is not a valid reference to a local units or a standard unit type.", validator->error(1)->description());
 }
 
+TEST(CoverageValidator, degreeElementWithOneSibling)
+{
+    const std::string math =
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
+        "  <apply>\n"
+        "    <root/>\n"
+        "    <degree><ci>n</ci></degree>\n"
+        "  </apply>\n"
+        "</math>\n";
+
+    auto parser = libcellml::Parser::create();
+    auto model = libcellml::Model::create("degree");
+
+    auto component = libcellml::Component::create("c");
+    auto variable = libcellml::Variable::create("n");
+    variable->setUnits("second");
+    component->addVariable(variable);
+    component->appendMath(math);
+
+    model->addComponent(component);
+
+    auto validator = libcellml::Validator::create();
+
+    validator->validateModel(model);
+    EXPECT_EQ(size_t(0), validator->errorCount());
+}
+
+TEST(CoverageValidator, invalidXmlIds)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("coverage/invalidxmlids.cellml"));
+
+    auto validator = libcellml::Validator::create();
+
+    validator->validateModel(model);
+    EXPECT_EQ(size_t(1), validator->errorCount());
+}
+
+TEST(CoverageValidator, componentMathWithRepeatedVariableNames)
+{
+    const std::string math =
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
+        "  <apply>\n"
+        "    <eq/>\n"
+        "    <ci>C</ci>\n"
+        "<!-- valid comment in math -->\n"
+        "    <apply>\n"
+        "      <plus/>\n"
+        "      <ci>A</ci>\n"
+        "      <ci id=\"ciid\" />\n"
+        "    </apply>\n"
+        "  </apply>\n"
+        "</math>\n";
+
+    libcellml::ValidatorPtr v = libcellml::Validator::create();
+    libcellml::ModelPtr m = libcellml::Model::create();
+    libcellml::ComponentPtr c = libcellml::Component::create();
+    libcellml::VariablePtr v1 = libcellml::Variable::create();
+    libcellml::VariablePtr v2 = libcellml::Variable::create();
+    libcellml::VariablePtr v3 = libcellml::Variable::create();
+
+    m->setName("modelName");
+    c->setName("componentName");
+    v1->setName("A");
+    v2->setName("B");
+    v3->setName("A");
+    v1->setInitialValue("1.0");
+    v2->setInitialValue("-1.0");
+    v1->setUnits("dimensionless");
+    v2->setUnits("dimensionless");
+    v3->setUnits("dimensionless");
+
+    c->addVariable(v1);
+    c->addVariable(v2);
+    c->addVariable(v3);
+    c->setMath(math);
+    m->addComponent(c);
+
+    v->validateModel(m);
+    EXPECT_EQ(size_t(3), v->issueCount());
```

Comment:
It would be nice to have the exact issue descriptions rather than just test for the number of issues.

Reply:
[528a80f](https://github.com/cellml/libcellml/pull/1128/commits/528a80f1273ba2e874ef174e26b25907d50a9688)